rust/src/import_scanning.rs

```diff
@@ -350,3 +350,76 @@ pub fn imports_by_module_to_py(
     }
     imports_by_module_py
 }
+
+/// Statically analyses the given module and returns a set of Modules that
+/// it imports.
+/// Python args:
+///
+/// - module_files                   The modules to scan.
+/// - found_packages:                Set of FoundPackages containing all the modules
+///                                  for analysis.
+/// - include_external_packages:     Whether to include imports of external modules (i.e.
+///                                  modules not contained in modules_by_package_directory)
+///                                  in the results.
+/// - exclude_type_checking_imports: If True, don't include imports behind TYPE_CHECKING guards.
+/// - file_system:                   The file system interface to use. (A BasicFileSystem.)
+///
+/// Returns dict[Module, set[DirectImport]].
+#[pyfunction]
+pub fn scan_for_imports<'py>(
```

Comment:
Good call. Originally the design was to have `lib.rs` as the Python layer but as you will have noticed, we are no longer sticking to that as the crate has grown.